### ISSUE TITLE
Calculate Number of memoryslots required  based on input schedules if not set.

### ISSUE THREAD

**Issue by taalexander** [Qiskit/qiskit-terra#2834]
### What is the expected enhancement?
At qobj assembly time the number of `memory_slots` required should be calculated from the input schedules if it is not set.

**Comment by ajavadia** [comment]
@Matt-Stypulkoski can you take this one after #2491 please?

**Comment by taalexander** [comment]
I have done this already in #2835 (forgot to # it). The same behavior could
be replicated for the qasm path though in a separate PR.
